client/reader/stats.js

```diff
@@ -22,20 +22,25 @@ export function recordGaEvent( action, label, value ) {
 	ga.recordEvent( 'Reader', action, label, value );
 }
 
-export function recordPermalinkClick( where, post ) {
+export function recordPermalinkClick( where, post, eventProperties = {} ) {
 	mc.bumpStat( {
 		reader_actions: 'visited_post_permalink',
 		reader_permalink_source: where,
 	} );
 	recordGaEvent( 'Clicked Post Permalink', where );
 	const trackEvent = 'calypso_reader_permalink_click';
-	const args = {
-		source: where,
-	};
+
+	// Add where as source prop
+	eventProperties = Object.assign( { source: where }, eventProperties );
+
+	// Add location as ui_algo prop
+	const location = getLocation();
```

Comment:
do you need to take care of `ui_algo` here since `recordTrack` will do it for you?

Reply:
Ah yes! Good point. Removing!